### ISSUE TITLE
Unwanted gimbal rotation with DJI Air 2S

### ISSUE THREAD

**Issue by naiveHobo** [dji-sdk/Mobile-SDK-Android#1215]
I have a custom DJI Mobile SDK application to control a DJI Air 2S. The Air 2S gimbal is expected to rotate in the pitch axis while having no rotation in the roll and yaw axes when the application receives a command.

During most flights the gimbal behaves as expected but there are certain flights every now and then when there is an unwanted gimbal rotation in the yaw axis. The gimbal ends up turning to far left or right such that the edges of the propeller guards on the drone are visible in the camera field of view. The gimbal seems to get stuck in that orientation even when other pitch rotation commands are sent to the application. The gimbal automatically gets back to the expected position (ie, no rotation in roll and yaw axes and expected angle in pitch axis) when I stop sending the drone any movement commands through the virtual sticks and get the drone to just hover.

What could be the issue here? This behavior happens erratically in maybe one of every 15-20 flights.

Here's the code snipped I use to initialize the gimbal when the application starts up after connecting to the product.
```java
gimbal.setPitchRangeExtensionEnabled(true, null);
gimbal.reset(Axis.ROLL, ResetDirection.CENTER, null);
gimbal.reset(Axis.PITCH, ResetDirection.CENTER, null);
gimbal.reset(Axis.YAW, ResetDirection.CENTER, null);
gimbal.setMode(GimbalMode.FPV, null);
```

Here's the code snipped used to set the camera angle in the pitch axis. This snippet is called as a callback while listening to any new gimbal rotation commands and the gimbal rotation commands are sent at about 10hz. So this snipped is called at 10hz with the required gimbal angle in the pitch axis.
```java
float pitch = 20.0f;
Rotation.Builder builder = new Rotation.Builder()
        .mode(RotationMode.ABSOLUTE_ANGLE)
        .pitch(pitch)
        .time(0.5);
gimbal.rotate(builder.build(), null);
```

**Comment by DJI-William** [comment]
Since you don't rotate the yaw angle, I will suggest you use the RotationMode in relative angle also with the gimbal mode of yaw_follow. I am not sure if the air2s does support the FPV gimbal mode as you set the callback as null. A possbile reason is you are using the absolute angle which means the yaw angle of 0 points to the True North.

**Comment by naiveHobo** [comment]
I understand, will try that out. Although, this behavior does seem to happen almost randomly. Most flights happen correctly without any unwanted yaw rotation, but every now and then the gimbal ends up yaw-ing unexpectedly.

Would I be able to use an absolute angle for only pitch rotation if the rotation mode is set to YAW_FOLLOW?

**Comment by DJI-William** [comment]
Yes, yaw_follow means the gimbal yaw will always follow the aircraft heading. It doesn not influence the pitch and roll.

**Comment by naiveHobo** [comment]
I see. I actually was using the yaw follow mode till some time ago but it resulted in pretty much the same behavior. Every now and then, the gimbal rotates in the yaw axis without any commands. Do you think it might have something to do with the frequency at which the `gimbal.rotate()` function is called? 

```java
Rotation.Builder builder = new Rotation.Builder()
        .mode(RotationMode.ABSOLUTE_ANGLE)
        .pitch(pitch)
        .time(0.5);
gimbal.rotate(builder.build(), null);
```

The time set in the rotation builder is set to 0.5s and the rotation commands are being issued at 10hz so this code snipped is called every 0.1 seconds asking the gimbal to rotate in 0.5 seconds. Does this somehow end up burdening the gimbal too much? The fact that it only happens randomly in about 1 of every 20 flights seems to suggest that it's not a code error I believe.

**Comment by DJI-William** [comment]
Shoudn't be the frequency. I cannot find anything wrong in the code. The only thing left is a bug then. How frequently this issue comes？

**Comment by naiveHobo** [comment]
I'd say once in every 20 flights approximately.

**Comment by Surriel** [comment]
@naiveHobo Hello, have you figured out how to resolve your issue? Im also currently facing this weird behavior every now and then. The gimbal yaw randomly misaligns sometime on YAW_FOLLOW. I would detect and attempt to realign but to no avail.
